src/doc_builder/commands/push.py

```diff
@@ -0,0 +1,208 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import base64
+from pathlib import Path
+from time import time
+from typing import List, Optional, TypedDict
+
+import requests
+from gql import Client, gql
+from gql.transport.aiohttp import AIOHTTPTransport
+
+
+def get_head_oid(repo_id: str, token: str, branch: Optional[str] = "main") -> str:
+    """
+    Returns last commit sha from repostory `repo_id` & branch `branch`
+    """
+    res = requests.get(
+        f"https://api.github.com/repos/{repo_id}/git/refs/heads/{branch}",
+        headers={"Authorization": f"bearer {token}"},
+    )
+    if res.status_code != 200:
+        raise Exception(f"get_head_oid failed: {res.message}")
+
+    res_json = res.json()
+    head_oid = res_json["object"]["sha"]
+    return head_oid
+
+
+class FileAddition(TypedDict):
+    path: str
+    contents: int
+
+
+def create_additions(library_name: str) -> List[FileAddition]:
+    """
+    Given `library_name` dir, returns [FileAddition!]!: [{path: "some_path", contents: "base64_repr_contents"}, ...]
+    see more here: https://docs.github.com/en/graphql/reference/input-objects#filechanges
+    """
+    p = Path(library_name)
+    files = [x for x in p.glob("**/*") if x.is_file()]
+    additions = []
+
+    for fpath in files:
+        with open(fpath, "rb") as reader:
+            content = reader.read()
+            content_base64 = base64.b64encode(content)
+            content_base64 = str(content_base64, "utf-8")
+            additions.append({"path": str(fpath), "contents": content_base64})
+
+    return additions
+
+
+MAX_CHUNK_LEN = 3e7  # 30 Megabytes
+
+
+def create_additions_chunks(additions: List[FileAddition]) -> List[List[FileAddition]]:
+    """
+    Github GraphQL `createCommitOnBranch` mutation fails when a payload is bigger than 50 MB.
+    Therefore, in those cases (transformers doc ~ 100 MB), we need to commit using
+    multiple smaller `createCommitOnBranch` mutation calls.
+    """
+    additions_chunks = []
+    current_chunk = []
+    current_len = 0
+
+    while additions:
+        addition = additions.pop()
```

Comment:
It doesn't look like `additions` is being modified in this loop, so let's just do a for loop:
```suggestion
    for addition in additions:
```

Reply:
great suggestion!